common/src/main/java/io/netty/util/internal/MaxDirectMemorySetting.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2013 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal;
+
+/**
+ * This class is the preferred way to set the max direct memory instead of using
+ * {@link PlatformDependent#JAVA_SYS_PROP_IO_NETTY_MAX_DIRECT_MEMORY}.
+ *
+ * Some infrastructure may prevent the use of Java system property from the user perspective, and configuration files
+ * allow to set the max direct memory programmatically using this class before any call made
+ * to {@link PlatformDependent}.
+ */
+public final class MaxDirectMemorySetting {
+
+  private static final long MAX_DIRECT_MEMORY_USE_DEFAULT = -1L;
+
+  private MaxDirectMemorySetting() {
+    throw new RuntimeException("By design this class cannot be instantiated.");
+  }
+
+  private static long maxDirectMemory = MAX_DIRECT_MEMORY_USE_DEFAULT;
+
+  public static synchronized void set(long m) {
+    if (m < 0) {
+      throw new IllegalArgumentException("Specified max_direct_memory must be greater or equal to 0, got " + m);
+    }
+
+    maxDirectMemory = m;
```

Comment:
I am not sure about this, i think the value will be assigned but it might not be used by the system if it is already set

Reply:
Yes, i think we should fail, i have added a check https://github.com/sirensolutions/netty/pull/25/commits/adac7ee842f9bd1100c4a4b8a2025e455006d894, thanks

Reply:
@scampi after cherry-picking this commit https://github.com/sirensolutions/netty/pull/25/commits/c4b48f53dbc3ee2ee821132298e01e7bd317a2c6 the check added here is always failing. From debugging it can be seen that the [MaxDirectMemorySetting.set(maxDirectMemory)](https://github.com/sirensolutions/siren-platform/blob/b45c373d9b1fe222ab3304e11bb61dd62fb38da1/core/src/main/java/io/siren/federate/core/SirenPlatformPlugin.java#L229) is called more than once hence the failure. I have removed the check in this commit https://github.com/sirensolutions/netty/pull/25/commits/113ca33be792366c974f763e11274e13028dfb0d